runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPoolFactory.java

```diff
@@ -0,0 +1,1095 @@
+
+    private final RequestHeaderFW requestHeaderRO = new RequestHeaderFW();
+    private final ResponseHeaderV0FW responseHeaderRO = new ResponseHeaderV0FW();
+
+    private final MergedBudgetCreditor creditor;
+    private final int proxyTypeId;
+    private final BufferPool bufferPool;
+    private final MutableDirectBuffer writeBuffer;
+    private final Signaler signaler;
+    private final BindingHandler streamFactory;
+    private final LongUnaryOperator supplyInitialId;
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+    private Object2ObjectHashMap<String, KafkaClientConnectionApp> connectionPool;
+
+    public KafkaClientConnectionPoolFactory(
+        KafkaConfiguration config,
+        EngineContext context,
+        LongFunction<KafkaBindingConfig> supplyBinding,
+        MergedBudgetCreditor creditor)
+    {
+        this.proxyTypeId = context.supplyTypeId("proxy");
+        this.writeBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
+        this.bufferPool = context.bufferPool();
+        this.signaler = context.signaler();
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
+        this.creditor = creditor;
+        this.connectionPool = new Object2ObjectHashMap();
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer sender)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long originId = begin.originId();
+        final long routedId = begin.routedId();
+        final long initialId = begin.streamId();
+        final long authorization = begin.authorization();
+        final long affinity = begin.affinity();
+
+        assert (initialId & 0x0000_0000_0000_0001L) != 0L;
+
+        final OctetsFW extension = begin.extension();
+        final ExtensionFW beginEx = extension.get(extensionRO::tryWrap);
+        assert beginEx != null && beginEx.typeId() == proxyTypeId;
+        final ProxyBeginExFW proxyBeginEx = extension.get(proxyBeginExRO::tryWrap);
+
+        MessageConsumer newStream = null;
+        String address = CLUSTER;
+        String host;
+        int port;
+
+        if (proxyBeginEx != null)
+        {
+            final ProxyAddressInetFW inet = proxyBeginEx.address().inet();
+            host = inet.destination().asString();
+            port = inet.destinationPort();
+            address = String.format("%s:%d", host, port);
+        }
+        else
+        {
+            host = null;
+            port = 0;
+        }
+
+        final KafkaClientConnectionApp kafkaClientConnectionApp = connectionPool.computeIfAbsent(address, s ->
```

Comment:
Rename `KafkaClientConnectionApp` to `KafkaClientConnection`.
Rename `kafkaClientConnectionApp` to `connection`.